src/function.rs

```diff
@@ -411,3 +415,14 @@ fn without_extension(_context: &FunctionContext, path: &str) -> Result<String, S
 
   Ok(parent.join(file_stem).to_string())
 }
+
+/// Check whether a string processes properly as semver (e.x. "0.1.0")
+/// and matches a given semver requirement (e.x. ">=0.1.0")
+fn semver_matches(_context: &FunctionContext, v: &str, req: &str) -> Result<String, String> {
+  Ok(
+    VersionReq::parse(req.as_ref())
+      .map_err(|err| err.to_string())?
```

Comment:
I'm curious what the error message looks like. Sometimes libraries produce errors which aren't super helpful when they're bubbled up to the end user. If it's ambiguous or doesn't give enough information, add context with something like:

```suggestion
      .map_err(|err| format!{"invalid semver: {err}")?
```